### ISSUE TITLE
Assigning new values to a protocol buffer frozen model in TensorFlow

### ISSUE THREAD

**Issue by amirjamez** [tensorflow/tensorflow#20194]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: 14.04
- **TensorFlow installed from (source or binary)**: Source
- **TensorFlow version (use command below)**: r1.8
- **Python version**:  2.7
- **Bazel version (if compiling from source)**: bazel release 0.13.0
- **GCC/Compiler version (if compiling from source)**: 4.8
- **CUDA/cuDNN version**: 8/7
- **GPU model and memory**: NVIDIA-1060
- **Exact command to reproduce**:

I am able to assign new values to a tensor by using `tf.assign` and `tf.get_collection(tf.GraphKeys.GLOBAL_VARIABLES)` when I restore a pretrained model:

```
 >>>sess.run(tf.assign([v for v in tf.get_collection(tf.GraphKeys.GLOBAL_VARIABLES) if v.name == 'LAYER_NAME'][0], New_Value_FOR_LAYER_NAME))`
```

However, in the case of a frozen model (`.pb`) file, the equivalent using `tf.assign` and `tf.get_default_graph().get_operations()` does not yeild any results:

```
>>>sess.run(tf.assign([v for v in tf.get_default_graph().get_operations() if v.name == LAYER_NAME'][0], New_Value_FOR_LAYER_NAME)). 
```
I guess the underlying reason is that pretrained models have [tf.Tensor](https://www.tensorflow.org/api_docs/python/tf/Tensor) and frozen models have [tf.Operation](https://www.tensorflow.org/api_docs/python/tf/Operation):

```
>>>[v for v in tf.get_default_graph().get_operations() if v.name == 'softmax/weights_quint8_const']
    [<tf.Operation 'softmax/weights_quint8_const' type=Const>]
```

Then, what is the correct way of assigning new values to a tf.Operation in a frozen model.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 32 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 46 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 76 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by amirjamez** [comment]
no answers on this?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 77 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by jvishnuvardhan** [comment]
Closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!

**Comment by ebrevdo** [comment]
For this it's better to modify the graphdef directly.  you'd export to a graph_def, modify that, then load it back in (or save it to file) depending on how you're going to use it

```
gd = tf.get_default_graph().as_graph_def()
const_nodes = [x for x in gd.node if x.op == 'Const']
value = const_nodes[0].attr['value'].tensor
# assuming an int32 vector
value.tensor_shape.dim[0].size = 3
value.tensor_content = np.array([1, 2, 3], dtype=np.int32).tostring()
# load gd back into a graph or store to disk
```

this will work for most data types, i think, including float and integer.  however it will break for string types.  presumably you're not freezing strings in graphs.
